src/StorageRegistry.sol

```diff
@@ -0,0 +1,321 @@
+
+    /// @dev Revert if the caller provides the wrong payment amount.
+    error InvalidPayment();
+
+    /// @dev Revert when there are not enough funds for a native token transfer.
+    error InsufficientFunds();
+
+    /// @dev Revert when a native token transfer fails.
+    error CallFailed();
+
+    /// @dev Revert if the price feed returns a stale price.
+    error StalePrice();
+
+    /// @dev Revert if the price feed round is incomplete and has not yet generated a price.
+    error IncompleteRound();
+
+    /// @dev Revert if the price feed returns a zero or negative price.
+    error InvalidPrice();
+
+    /// @dev Revert if the sequencer uptime feed detects that the L2 sequencer is unavailable.
+    error SequencerDown();
+
+    /// @dev Revert if the L2 sequencer restarted less than L2_DOWNTIME_GRACE_PERIOD seconds ago.
+    error GracePeriodNotOver();
+
+    /*//////////////////////////////////////////////////////////////
+                                 EVENTS
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @dev Emit an event when caller pays rent for an fid's storage.
+     *
+     * @param payer     Address of the account paying the storage rent.
+     * @param fid       The fid that will receive the storage allocation.
+     * @param units     The number of storage units being rented.
+     */
+    event Rent(address indexed payer, uint256 indexed fid, uint256 units);
+
+    /**
+     * @dev Emit an event when an owner changes the price of storage units.
+     *
+     * @param oldPrice The previous unit price in USD. Fixed point value with 8 decimals.
+     * @param newPrice The new unit price in USD. Fixed point value with 8 decimals.
+     */
+    event SetPrice(uint256 oldPrice, uint256 newPrice);
+
+    /**
+     * @dev Emit an event when an owner changes the maximum supply of storage units.
+     *
+     * @param oldMax The previous maximum amount.
+     * @param newMax The new maximum amount.
+     */
+    event SetMaxUnits(uint256 oldMax, uint256 newMax);
+
+    /**
+     * @dev Emit an event when an owner changes the deprecationTimestamp.
+     *
+     * @param oldTimestamp The previous deprecationTimestamp.
+     * @param newTimestamp The new deprecationTimestamp.
+     */
+    event SetDeprecationTimestamp(uint256 oldTimestamp, uint256 newTimestamp);
+
+    /**
+     * @dev Emit an event when an owner makes a withdrawal from the contract balance.
+     *
+     * @param to     Address of recipient.
+     * @param amount The amount of ether withdrawn.
+     */
+    event Withdraw(address indexed to, uint256 amount);
+
+    /*//////////////////////////////////////////////////////////////
+                                CONSTANTS
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @dev Period in seconds to wait after the L2 sequencer restarts before resuming rentals.
+     */
+    uint256 public constant L2_DOWNTIME_GRACE_PERIOD = 3600;
```

Comment:
Would be helpful to have some comments here about why this number is the right tradeoff for us

Reply:
Decided to parameterize this rather than use a constant because I don't have a great answer. Seems like something we might want the ability to change. This is the value used in the Chainlink docs—I'll read up on this further.